postgres/datadog_checks/postgres/statements.py

```diff
@@ -157,33 +174,20 @@ def _collect_statement_metrics(self, instance_tags):
             return
 
         cursor = self.db.cursor(cursor_factory=psycopg2.extras.DictCursor)
-        columns = []
-        for entry in (PG_STAT_STATEMENTS_METRIC_COLUMNS, PG_STAT_STATEMENTS_TAG_COLUMNS):
-            for alias, (column, *_) in entry.items():
-                # Only include columns which are available on the table
-                if alias in self._pg_stat_statements_columns or alias in PG_STAT_STATEMENTS_TAG_COLUMNS:
-                    columns.append('{column} AS {alias}'.format(column=column, alias=alias))
-
         rows = self._execute_query(cursor, STATEMENTS_QUERY.format(
-            cols=', '.join(columns),
+            cols=', '.join(self._pg_stat_statements_query_columns),
             pg_stat_statements_function=self.config.pg_stat_statements_function
         ), params=(self.config.dbname,))
+        statsd.gauge("dd.postgres.collect_statement_metrics.rows", len(rows), tags=instance_tags)
         if not rows:
             return
         rows = rows[:self.config.max_query_metrics]
+        row_keyfunc = lambda row: (row['queryid'], row['query'])
```

Comment:
This should be `queryid`, `datname`, `rolname`

Reply:
Good catch. Totally missed rolname. I figured datname didn't matter because it was always the same database, but might as well add it.

Reply:
I think pg_stat_statements is global though, so it will show queries from all databases

Reply:
We filter in the query though `WHERE pg_database.datname = %s`